pkg/descheduler/controllers/migration/arbitrator/sort.go

```diff
@@ -0,0 +1,191 @@
+/*
+Copyright 2023 The Koordinator Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package arbitrator
+
+import (
+	"context"
+	"math"
+	"sort"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	v1alpha1 "github.com/koordinator-sh/koordinator/apis/scheduling/v1alpha1"
+	"github.com/koordinator-sh/koordinator/pkg/descheduler/fieldindex"
+	"github.com/koordinator-sh/koordinator/pkg/descheduler/utils/sorter"
+	utilclient "github.com/koordinator-sh/koordinator/pkg/util/client"
+)
+
+const (
+	JobKind = "Job"
+)
+
+// NewPodSortFn sort a SortFn that sorts PodMigrationJobs by their Pods, including priority, QoS.
+func NewPodSortFn(sorter *sorter.MultiSorter) SortFn {
+	return func(jobs []*v1alpha1.PodMigrationJob, podOfJob map[*v1alpha1.PodMigrationJob]*corev1.Pod) []*v1alpha1.PodMigrationJob {
+		var pods []*corev1.Pod
+		jobOfPod := map[*corev1.Pod]*v1alpha1.PodMigrationJob{}
```

Comment:
```suggestion
func PodSort(sorter func(pods []*corev1.Pod)) SortFn {
	return func(jobs []*v1alpha1.PodMigrationJob, podOfJob map[*v1alpha1.PodMigrationJob]*corev1.Pod) []*v1alpha1.PodMigrationJob {
		var pods []*corev1.Pod
		jobOfPod := map[*corev1.Pod]*v1alpha1.PodMigrationJob{}
```
Here we can remove the dependency of the sorter package and replace it with higher-order functions.
And we can simplify the function name, you can refer to the sorter package

Reply:
ok